src/store/setting.ts

```diff
@@ -309,50 +318,66 @@ export const settingStore = createPartialStore<SettingStoreTypes>({
     },
   },
 
+  SET_ENGINE_SETTING: {
+    mutation(state, { engineSetting, engineId }) {
+      state.engineSetting[engineId] = engineSetting;
+    },
+    async action({ commit }, { engineSetting, engineId }) {
+      window.electron.setSetting("engineSetting", {
+        ...(await window.electron.getSetting("engineSetting")),
+        [engineId]: engineSetting,
+      });
+      commit("SET_ENGINE_SETTING", { engineSetting, engineId });
+    },
+  },
+
   CHANGE_USE_GPU: {
     /**
      * CPU/GPUモードを切り替えようとする。
      * GPUモードでエンジン起動に失敗した場合はCPUモードに戻す。
      */
-    action: createUILockAction(async ({ state, dispatch }, { useGpu }) => {
-      if (state.useGpu === useGpu) return;
-
-      const isAvailableGPUMode = await window.electron.isAvailableGPUMode();
-
-      // 対応するGPUがない場合に変更を続行するか問う
-      if (useGpu && !isAvailableGPUMode) {
-        const result = await window.electron.showQuestionDialog({
-          type: "warning",
-          title: "対応するGPUデバイスが見つかりません",
-          message:
-            "GPUモードの利用には対応するGPUデバイスが必要です。\n" +
-            "このままGPUモードに変更するとエンジンエラーが発生する可能性があります。本当に変更しますか？",
-          buttons: ["変更する", "変更しない"],
-          cancelId: 1,
-        });
-        if (result == 1) {
-          return;
+    action: createUILockAction(
+      async ({ state, dispatch }, { useGpu, engineId }) => {
+        const isAvailableGPUMode = await window.electron.isAvailableGPUMode();
+
+        // 対応するGPUがない場合に変更を続行するか問う
+        if (useGpu === true && !isAvailableGPUMode) {
+          const result = await window.electron.showQuestionDialog({
+            type: "warning",
+            title: "対応するGPUデバイスが見つかりません",
+            message:
+              "GPUモードの利用には対応するGPUデバイスが必要です。\n" +
+              "このままGPUモードに変更するとエンジンエラーが発生する可能性があります。本当に変更しますか？",
+            buttons: ["変更する", "変更しない"],
+            cancelId: 1,
+          });
+          if (result == 1) {
+            return;
+          }
         }
-      }
 
-      await dispatch("SET_USE_GPU", { useGpu });
-      const success = await dispatch("RESTART_ENGINES", {
-        engineIds: state.engineIds,
-      });
-
-      // GPUモードに変更できなかった場合はCPUモードに戻す
-      // FIXME: useGpu設定を保存してからエンジン起動を試すのではなく、逆にしたい
-      if (!success && useGpu) {
-        await window.electron.showMessageDialog({
-          type: "error",
-          title: "GPUモードに変更できませんでした",
-          message:
-            "GPUモードでエンジンを起動できなかったためCPUモードに戻します",
+        await dispatch("SET_ENGINE_SETTING", {
+          engineSetting: { ...state.engineSetting[engineId], useGpu },
+          engineId,
+        });
+        const result = await dispatch("RESTART_ENGINES", {
+          engineIds: [engineId],
         });
-        await dispatch("CHANGE_USE_GPU", { useGpu: false });
-        return;
+
+        // GPUモードに変更できなかった場合はCPUモードに戻す
+        // FIXME: useGpu設定を保存してからエンジン起動を試すのではなく、逆にしたい
+        if (!result && useGpu) {
```

Comment:
こうかも（以前のコードからバグってそう）
```suggestion
        if (!result.success && useGpu) {
```